atlas/AtlasMl/atlasml/routers/competency.py

```
@@ -1,143 +1,137 @@
+            validated_description, course_id=validated_course_id
+        )
+
+        logger.info(f"Successfully suggested {len(competencies)} competencies")
+        return SuggestCompetencyResponse(competencies=competencies)
+
+    except HTTPException:
+        # Re-raise HTTP exceptions
+        raise
+    except Exception as e:
+        # Use centralized error handling
+        raise handle_pipeline_error(e, "suggest_competencies")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)